addons/survey/models/survey_survey.py

```diff
@@ -401,9 +410,24 @@ def create(self, vals_list):
         return surveys
 
     def write(self, vals):
+        changing_speed_rating_surveys = self.browse()
+        new_speed_rating = vals.get('session_speed_rating')
+        if new_speed_rating is not None:
+            changing_speed_rating_surveys |= self.filtered(lambda s: s.session_speed_rating != new_speed_rating)
+        new_session_speed_rating_time_limit = vals.get('session_speed_rating_time_limit')
+        if new_session_speed_rating_time_limit is not None:
+            changing_speed_rating_surveys |= self.filtered(lambda s: s.session_speed_rating_time_limit != new_session_speed_rating_time_limit)
```

Comment:
Proposal to filter once (and keep ordering, allowing to avoid non deterministic stuff)
```
speed_rating, speed_limit = vals.get('session_speed_rating'), vals.get('session_speed_rating_time_limit')
domain = expression.OR([
    [('session_speed_rating', '!=', speed_rating)] if speed_rating else [],
    [('session_speed_rating_time_limit', '!=', speed_limit)] if speed_limit else [],
])
surveys_toupdate = self.filtered_domain(domain) if domain else self.browse()
```